main.go

```diff
@@ -0,0 +1,145 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+
+	"golang.org/x/mod/modfile"
+	"golang.org/x/mod/semver"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	nvmInstallMessage = "You can download the required Node.js version using Node Version Manager (https://github.com/nvm-sh/nvm). Once nvm is installed, run `nvm install` in this directory to install the correct Node version."
+)
+
+var mmplugCmd = &cobra.Command{
+	Use:   "mmplug",
+	Short: "mmplug is a command line tool to manage plugin projects",
+}
+
+var doctorCmd = &cobra.Command{
+	Use:   "doctor",
+	Short: "Check if your development environment set up correctly",
+	Run:   runDoctor,
+}
+
+func main() {
+	mmplugCmd.AddCommand(doctorCmd)
+
+	if err := mmplugCmd.Execute(); err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		os.Exit(1)
+	}
+}
+
+func runDoctor(cmd *cobra.Command, args []string) {
+	fmt.Printf("Checking project dependencies\n")
+
+	allPassed := true
+
+	fmt.Println()
+	if !runGoCheck() {
+		allPassed = false
+	}
+
+	fmt.Println()
+	if !runNodeCheck() {
+		allPassed = false
+	}
+
+	fmt.Println()
+	if !allPassed {
+		fail("Not all checks passed. Please fix the above issues and try again.")
+	} else {
+		success("All checks passed.")
+	}
+}
+
+func runGoCheck() bool {
+	goModFile, err := ioutil.ReadFile("go.mod")
```

Comment:
same as .nvrm, I would take go.mod as the trigger to check go version against it. For a webapp-only plugin, we shouldn't force users to have go.mod.